--- ffmpeg-4.0/libavfilter/Makefile-	2018-04-20 13:02:57.000000000 +0300
+++ ffmpeg-4.0/libavfilter/Makefile	2018-05-05 22:30:54.670860702 +0300
@@ -106,6 +106,12 @@ OBJS-$(CONFIG_LOWPASS_FILTER)
 OBJS-$(CONFIG_LV2_FILTER)                    += af_lv2.o
 OBJS-$(CONFIG_MCOMPAND_FILTER)               += af_mcompand.o
 OBJS-$(CONFIG_PAN_FILTER)                    += af_pan.o
+OBJS-$(CONFIG_QADRC_FILTER)                  += af_qadrc.o
+OBJS-$(CONFIG_QALIMITER_FILTER)              += af_qalimiter.o
+OBJS-$(CONFIG_MYDRC_FILTER)                  += af_mydrc.o
+OBJS-$(CONFIG_MONOPARTS_FILTER)              += af_monoparts.o
+libavfilter/af_qadrc.o libavfilter/af_qalimiter.o libavfilter/af_mydrc.o libavfilter/af_monoparts.o: \
+CFLAGS += -g -ffast-math -ftree-vectorize -fopt-info-vec -Wno-declaration-after-statement
 OBJS-$(CONFIG_REPLAYGAIN_FILTER)             += af_replaygain.o
 OBJS-$(CONFIG_RESAMPLE_FILTER)               += af_resample.o
 OBJS-$(CONFIG_RUBBERBAND_FILTER)             += af_rubberband.o
--- ffmpeg-4.0/libavfilter/allfilters.c-	2018-05-05 22:30:54.670860702 +0300
+++ ffmpeg-4.0/libavfilter/allfilters.c	2018-05-05 22:33:49.351575649 +0300
@@ -101,6 +101,10 @@ extern AVFilter ff_af_lowpass;
 extern AVFilter ff_af_lv2;
 extern AVFilter ff_af_mcompand;
 extern AVFilter ff_af_pan;
+extern AVFilter ff_af_qadrc;
+extern AVFilter ff_af_qalimiter;
+extern AVFilter ff_af_mydrc;
+extern AVFilter ff_af_monoparts;
 extern AVFilter ff_af_replaygain;
 extern AVFilter ff_af_resample;
 extern AVFilter ff_af_rubberband;
